File: backend/internal/domain/store.go
package models

import (
	"time"
)

type Store struct {
	ID            string    `gorm:"type:uuid;default:uuid_generate_v4();primaryKey"`
	Name          string    `gorm:"not null" json:"name"`
	Address       string    `gorm:"type:text" json:"address"`
	PhoneNumber   string    `json:"phone_number"`
	BusinessHours string    `json:"business_hours"`
	ZipCode       string    `json:"zip_code"`
	Description   string    `gorm:"type:text" json:"description"`
	IsActive      bool      `gorm:"default:true" json:"is_active"`
	CreatedBy     string    `gorm:"type:uuid" json:"created_by"`
	UpdatedBy     string    `gorm:"type:uuid" json:"updated_by"`
	CreatedAt     time.Time `json:"created_at"`
	UpdatedAt     time.Time `json:"updated_at"`
}

File: frontend/config/types/api/store.ts
export type StoreStatus = "active" | "inactive" | "deleted";

// 基本の店舗情報
export interface Store {
  id: string;
  name: string;
  address: string;
  phoneNumber: string;
  businessHours: BusinessHours;
  isActive: boolean;
  description?: string;
  imageUrl?: string;
  createdBy: string;
  updatedBy: string;
  createdAt: string;
  updatedAt: string;
}

// 営業時間の型定義
export interface BusinessHours {
  start: string; // HH:mm形式
  end: string; // HH:mm形式
  regularHoliday?: string[]; // 定休日（曜日）
}

// 店舗作成・更新時のリクエスト型
export interface CreateStoreRequest {
  name: string;
  address: string;
  phoneNumber: string;
  businessHours: BusinessHours;
  description?: string;
  imageUrl?: string;
}

export interface UpdateStoreRequest extends Partial<CreateStoreRequest> {
  id: string;
}

// フィルタリング用の型
export interface StoreFilter {
  search?: string;
  isActive?: boolean;
  page?: number;
  limit?: number;
  sortBy?: "name" | "createdAt";
  sortOrder?: "asc" | "desc";
}

// フォーム用の型
export interface StoreFormData extends Omit<Store, "id"> {
  id?: string;
}

// APIレスポンス用
export interface StoresApiResponse {
  stores: Store[];
  total: number;
  page: number;
  limit: number;
}

export type StoreResponse = Store;
export type StoresResponse = Store[];
